### PR TITLE
feature #6250 - added xdai and poa sidechains

### LINKED ISSUE #6250: Add Custom RPC Providers for POA and DAIChain and change default currencies
### User Story

The developers from POA want us to add some custom RPC endpoints to our `Network settings` screen, so that people at #CryptoLife can easily connect to their chains from Status.

### Description

*Type*: Feature

*Summary*: POA has already tested the endpoints they provided us with, and says most things work as they should, they just need us to add those RPC endpoints to the above screen by default and make UI changes, especially around the default currency and terminology when users are switched to those chains. 

We should also update this for the Ropsten and Rinkeby testnets, as it is a nice idea in general for usability, i.e. ETHro and ETHri.

#### Expected behavior

Ability to connect to custom RPCs easily by default and have the UI change seamlessly to reflect this different network, especially w.r.t the currencies displayed in our wallet.

### Additional Information

The RPC endpoints they have provided for us to test against are as follows:

xDai Chain: https://dai.poa.network (network id 100)
POA Network: https://poa.infura.io (network id 99)

<blockquote></blockquote>
<blockquote></blockquote>

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #6250 

### Summary:

- Adds two custom RPC networks for POA Network and xDAI Chain
- Adds custom branding for native currencies in those two networks (name, symbol, icon)
- Adds support for fiat prices of POA and xDAI (using POA and DAI prices from cryptocompare, resp.)

### Testing notes (optional):
Note: after this change fiat value of ETHro and ETHri will correctly be calculated as zero.

#### Areas that maybe impacted (optional)
**Functional**

- wallet / transactions
- networks

### Steps to test:
- Switch to new networks and check if native currency is displayed correctly.
- Try sending and receiving POA and xDAI on their respective networks

status: ready

### CODE REVIEW

**Comment by goranjovic** [comment]
screenshot-2018-10-18_13 55 51 688

screenshot-2018-10-18_13 26 14 727

screenshot-2018-10-18_13 26 40 614

**Comment by goranjovic** [comment]
@andytudhope @SiddigZeidan we're going to need some POA and xDAI to test this

**Comment by SiddigZeidan** [comment]
@goranjovic please send your test address and we'll send some POA and xDai to the same account for you to test

File: src/status_im/protocol/core.cljs [hunk 1]
```diff
@@ -74,5 +74,6 @@
                                (not= network-id fetched-network-id))
                       (utils/show-popup
                        "Ethereum node started incorrectly"
-                       "Ethereum node was started with incorrect configuration, application will be stopped to recover from that condition."
+                       (str "Ethereum node was started with incorrect configuration, application will be stopped to recover from that condition."
```
Comment (rasom) [inline src/status_im/protocol/core.cljs]: Should it be moved to translations?

**Comment by goranjovic** [comment]
@SiddigZeidan sent by email

**Review by dmitryn — APPROVED** [review]

**Comment by annadanchenko** [comment]
@goranjovic I've recovered account with transactions done on mainnet, ropsten and rinkeby, switched to POA network and can see all transactions from mainnet but with POA as currency (there were ETH on mainnet). Expected: no transactions shown. Android device
If switch to Rinkeby then see expected ETHri and tokens transactions fine, so issue seems to be only with 2 new netwroks.
Issue on POA:
screenshot_20181018-181004
Same issue for xDai.
screenshot_20181018-181447

**Review by janherich — APPROVED** [review]

**Comment by annadanchenko** [comment]
@goranjovic Issue 2: Can't send by scanning QR: if scan QR code from Recieve transaction then error "Network does not match..." is shown. Both contacts are on POA network.

Steps:
1. switch to POA as user A and B
2. user A: Wallet->Send transaction-> Choose REcipient->Scan QR code -> scan code from Wallet ->Receive transaction of user B
screenshot_20181019-105452

Issue 3: "You are on the Mainnet..." label is shown on Send Transaction screen when POA network is selected.

screenshot_20181019-105401

**Comment by annadanchenko** [comment]
Issue 4: there is no fiat price shown for POA in Wallet and in 1:1 chat. Wallet shows "..." and 1:1 chat shows 0 USD as price

**Comment by annadanchenko** [comment]
Issue 4: fiat price in Wallet is shown as "..." for POA and xDAI. In 1:1 chat it's shown as 0 USD. Expected: some real fiat price value.
screenshot_20181019-111430

Issue 5: Only for xDAI: default gas price is set to 0 wei and it's possible to send such tx. Expected: gas price is 1 wei or more
screenshot_20181019-112002

**Comment by goranjovic** [comment]
@SiddigZeidan @adambabik  `web3.getGasPrice` always gives us zero for xDAI chain. It looks good for POA. Any ideas?

**Comment by goranjovic** [comment]
@rasom updated the i18n string

**Comment by goranjovic** [comment]
@annadanchenko pushed fixed for all the issues except the default gas for xdai

**Comment by annadanchenko** [comment]
@goranjovic @jakubgs build failed

**Comment by goranjovic** [comment]
@annadanchenko here's a new one ^

**Comment by annadanchenko** [comment]
@goranjovic it looks better, all issues besides 0 wei are almost done. Probably because PR is not rebased I still see issue fixed in https://github.com/status-im/status-react/pull/6404 but it does not block me.

New Issue 6: 0 USD instead of fiat value is shown in request message in 1:1 chat (for both recipient and sender) if both contacts are on xDai (or both are on POA). Does not matter if it's sent from wallet or in 1:1 chat /request command. Same for POA. Expected: fiat value is shown instead of 0.  E.g. 1 USD for 1 xDAI
Steps:
1. sender and recipient are on xDAI
2. sender sends request of 10 DAI to recipient in 1:1 chat (or in wallet)
3. as recipient and sender check fiat value for 10 requested DAI in 1:1 chat
Note: send message shows non-zero fiat value fine, so issue is only for request
screenshot_20181019-165626
screenshot_20181021-182905

**Comment by annadanchenko** [comment]
@goranjovic Issue 7: wrong fiat non-zero value is shown for test ETH (ETHro, ETHri) if sender is on test network and recipient is on non-test network (Mainnet, xDai or POA)
Steps:
1, Recipient is on mainnet
2. sender on Rinkeby in 1:1 chat sends 0.0003 ETHri
3. check fiat value on both sides shown in 1:1 chat for this tx 
Expected: on both side 0 USD is shown for ETHri fiat value
screenshot_20181019-175925

**Comment by SiddigZeidan** [comment]
> @SiddigZeidan @adambabik `web3.getGasPrice` always gives us zero for xDAI chain. It looks good for POA. Any ideas?

Goran, I asked one of our developers who reckons it could be the last few blocks have had no transactions and as such the `web3.getGasPrice` will give zero. 

"Docs tell that the gas price is determined by the last few blocks median gas price: https://web3js.readthedocs.io/en/1.0/web3-eth.html#getgasprice"

He will look into it further and reply

<blockquote><div><strong><a href="https://web3js.readthedocs.io/en/1.0/web3-eth.html#getgasprice">web3.eth — web3.js 1.0.0 documentation</a></strong></div></blockquote>

**Comment by annadanchenko** [comment]
Tests to do after fixes for the rest of issues:
1. from wallet and from 1:1 chat, on all networks with ETH and with token where possible:
1.1 Send from wallet
1.2 Send from 1:1 chat
1.3 request from 1:1 chat
1.4 request from wallet
1.5 reply to request with send 
To check: currency is ETH, ETHri, ETHro, POA, xDAI. Fiat value is non-zero only on mainnet, POA and xDAI. Check in Wallet screens, tx history, 1:1 chat on both recipient and sender sides
2. Wallet: total fiat value, value of asset: non-zero for mainnet, POA and xDAI. For ropsten and rinkevy: 0.
3. Scan QR code from Recieve transaction screen and send it - check tx is success, check values and currency shown
4. Sender and recipient are on different networks - see how request, send are shown in 1:1 chat
5. TX history will be empty for DAI and POA, only current tx is shown there but will disappear - it's expected, because they are not on etherscan
6. Upgrade: after upgrade tx sent before upgrade are shown with ETHRi (on Rinkeby), ETHro (on Ropsten) and have fiat zero value, same for requests. On mainnet: ETH and non-zero value.
7. Upgrade: after upgrade user can switch to POA and to DAI and perform tx there. Tx history does not show transactions.

**Comment by annadanchenko** [comment]
@goranjovic Issue 8: after upgrade from 0.9.29 to PR build from https://github.com/status-im/status-react/pull/6388#issuecomment-431377632, there are no new networks POA and DAI (both iOS and Android). Network settings shows Mainnet, Ropsten and Rinkeby as on 0.9.29. 
Expected: after user upgraded from 0.9.29 to 0.9.30 there are 2 new networks in the list

Android 0.9.29 release apk: https://drive.google.com/open?id=1PHnITeXCp3DZGhmhq0SMhjTgcOoUDBe8

**Comment by annadanchenko** [comment]
@goranjovic Issue 9: wrong fiat value of ETH ~0.22 USD is shown if recipient is on POA network and request sender is on mainnet. Expected: ETH fiat value on mainnet is shown or some note like can't calculate.

Steps:
1. recipient is on POA
2. sender is on mainnet
3. sender sends request in 1:1 chat of 2 ETH to recipient
4. as recipient check fiat value for sent message in 1:1 chat
screenshot_20181021-175606

**Comment by goranjovic** [comment]
Doc all the steps to add a network - https://discuss.status.im/t/dev-tutorial-how-to-add-new-network/598

<blockquote><div> Status.im</div><div><strong><a href="https://discuss.status.im/t/dev-tutorial-how-to-add-new-network/598">Dev Tutorial - How To Add New Network</a></strong></div><div>This doc lists and explains all the steps needed to add a new network to Status (e.g. side chain or a test network).  Inclusion of POA Network and xDAI Chain was the feature that led to this document, so I’ll use them as an example throughout this doc.  Overview:   add network info to status-im.constants  add chain id and name to status-im.utils.ethereum.core  add native currency info to status-im.utils.ethereum.tokens  add native currency icon to resources/images/tokens/NETWORK/0-native.png  ad...</div></blockquote>

**Comment by Serhy** [comment]
Hey @goranjovic. Jenkins build job keep failing to finish, but iOS (https://i.diawi.com/h2VocR)
and Android (https://ci.status.im/job/status-react/job/combined/job/mobile-android/5687/artifact/StatusIm-181022-134340-2c640a-pr.apk) builds were created.

I've checked all above nine issues @annadanchenko reported:

The `Issue 6` (https://github.com/status-im/status-react/pull/6388?nocache=fff%D1%83%D1%83#issuecomment-431393271) is still taking place: no fiat value in`/request` messages for POA / xDAI sent from Wallet or directly in 1-1 chat

Upgrade from Release 0.9.29 (or, for example, from 2 days old develop (nightly 19th of Octobe)) to latest build (`Issue 8` above) constantly fires up 'We were not able to decrypt your data'. `Cancel` / `Apply` flushes the data in app so all contacts are cleared. Always reproducible on Android / iOS:
**To reproduce**: 
1) Install Release 0.9.29
2) Create new account
3) Upgrade to https://ci.status.im/job/status-react/job/combined/job/mobile-android/5687/artifact/StatusIm-181022-134340-2c640a-pr.apk app version (https://i.diawi.com/h2VocR iOS version accordingly)

<blockquote><div><strong><a href="https://i.diawi.com/h2VocR">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote></blockquote>

**Comment by Serhy** [comment]
One more issue 10: When app without code changes this PR has receives `Request POA` / `Request xDAI` message - app crashed with 'Error:  Doesn't support name:` messaeg (and there is no way to log in to this account anymore until affected account upgrade the app where POA/xDAI changes are present OR restore account (however, please note in 0.9.30 we pull the 24h history - issue https://github.com/status-im/status-react/issues/6417 so in that case account will need to wit 24h)

Steps to reproduce:
1) Recover an account with xDAI / POA (the build mentioned in https://github.com/status-im/status-react/pull/6388?nocache=fff%D1%83%D1%83#issuecomment-431903594) on Device1 -> `UserxDAI-POA`
2) Create new account on app version without xDAI/POA changes but later than 0.9.29 release (e.g. latest nightly build) - `UserB`
3) `UserxDAI-POA` adds `UserB` in contacts and sends 'Request xDAI' / `Request POA` 

**Actual result:**  Android:
screen shot 2018-10-22 at 18 44 50
Desktop:
screen shot 2018-10-22 at 21 03 15

NOTE: it's fine for `/Send` messages to older app versions
NOTE2: it's all fine in 0.9.29 Release.

This is what happened for my account which I always use on Desktop. I used it to send / receive xDAI/POA on Android and iOS. Then I opened Desktop app where the same account existed (desktop nightly build) and seen above message.

**Comment by goranjovic** [comment]
@Serhy I had the same issue with upgrade in dev mode, but couldn't get release builds to reproduce them there too.

Looking into the error message.

**Comment by goranjovic** [comment]
@Serhy @asemiankevich I pushed a fix that should prevent this the next time we add new networks or tokens. It doesn't fix messages from PR to nightly, but messages from PR to 0.9.29 should work ok.

**Comment by asemiankevich** [comment]
@goranjovic is last build okay to test or i need to wait for another one?

**Comment by goranjovic** [comment]
@asemiankevich the new one is better :(

**Comment by asemiankevich** [comment]
@goranjovic so in last build https://github.com/status-im/status-react/pull/6388#issuecomment-432140959:

- when upgrading from release 0.9.29 to current PR: `We were not able to decrypt your data`
- when sending xDAI request from this PR to current nightly: `Error does not support name`
- fiat value for `/request` messages of DAI / POA is not shown in 1x1 chat

All issues @Serhy mentioned are not fixed in last build. Checking previous scenarios if they are not broken

**Comment by annadanchenko** [comment]
@goranjovic same issue with "We are not able to decrypt your data" on the new build 13 Android from https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181023-101801-399690-pr.apk 

<blockquote></blockquote>

**Comment by goranjovic** [comment]
@annadanchenko the issue looks related to the migration, looking into it

**Comment by goranjovic** [comment]
@annadanchenko just pushed the change that removes the migration to unblock the release. what this means is that users who want to use the sidechains will need to create a new account (not upgrade existing), but at least they'll have those chains.

cc @chadyj @andytudhope

**Comment by annadanchenko** [comment]
@goranjovic 

### Fixed
1. Upgrade issue `We were not able to decrypt your data` is gone. For upgraded account, no POA/xDAI networks shown - now expected. User who upgraded can create a second account and there are new networks.
2. Issue 7: wrong fiat non-zero value is shown for test ETH (ETHro, ETHri) if sender is on test network and recipient is on non-test network (Mainnet, xDai or POA) - now no fiat value is shown if there is network mismatch.
3. Issue 9: wrong fiat value of ETH ~0.22 USD is shown if recipient is on POA network and request sender is on mainnet. - now no fiat value is shown if there is network mismatch.

### Ignored
1. Issue 10 (crash if get request of POA on build after 0.9.29) can be fixed by updating to this PR (so assume also by updating to release build with this PR), so not a blocker.

### Still Open
1. Issue 6 still occurs: request message shows 0 USD for POA/xDAI https://github.com/status-im/status-react/pull/6388#issuecomment-431393271

### New 
1. Issue 11: non-zero fiat value for ETHro in sent message is shown if recipient was on mainnet when this message was received. 
Note: If both recipient and sender are on ropsten when message is sent then expected 0 fiat value is shown.

Steps:
1. Recipient is on Mainnet, sender is on Ropsten
2. Sender sends 0.08 ETHro in 1:1 to recipient
3. Recipient switches to Ropsten 
4. As recipient check fiat value shown, in my case it's 13.99 EUR







will update as I test more

**Comment by SiddigZeidan** [comment]
@goranjovic , our developer has created an issue for Parity to address zero gasPrice. Is it essential that issue gets fixed in order implement the xDai Chain?

https://github.com/paritytech/parity-ethereum/issues/9795

**Comment by annadanchenko** [comment]
known open issues are not critical and we can merge this PR to the release + document know issues in notes:
1. No POA and xDAI networks if upgrade
2. if build after 0.9.29 release was used and POA/DAI request is received then you will have constant 'Error: Doesn't support name:`. Can be fixed by upgrading to 0.9.30 release
3. Transaction history is not shown for POA and DAI
4. request message shows 0 fiat value for POA/xDAI
5. non-zero fiat value for ETHro in sent message is shown if recipient was on mainnet when this message was received.
6. on DAI sometimes default gas price is 0 wei. Can be fixed by manually setting gas price in Wallet -> Send transaction -> Advanced

**Comment by chadyj** [comment]
Thanks Anna. Updated the release notes.

**Comment by chadyj** [comment]
Cheers Goran! If you haven't already can you please cherry pick for release?

**Comment by goranjovic** [comment]
@SiddigZeidan Not a blocker - if that does happen users can always set the gas price themselves. Besides, if absence of transaction traffic is the problem, I reckon we'll generate enough during cryptolife to get the prices

**Comment by goranjovic** [comment]
@chadyj done

**Pull request merged**